tests/common.cpp

```diff
@@ -42,6 +43,203 @@ namespace gdrcopy {
             }
         }
 
+        CUresult gpu_mem_alloc(gpu_mem_handle_t *handle, const size_t size, bool aligned_mapping, bool set_sync_memops)
+        {
+            CUresult ret = CUDA_SUCCESS;
+            CUdeviceptr ptr, out_ptr;
+            size_t allocated_size;
+
+            if (aligned_mapping)
+                allocated_size = size + GPU_PAGE_SIZE - 1;
+            else
+                allocated_size = size;
+
+            ret = cuMemAlloc(&ptr, allocated_size);
+            if (ret != CUDA_SUCCESS)
+                return ret;
+
+            if (set_sync_memops) {
+                unsigned int flag = 1;
+                ret = cuPointerSetAttribute(&flag, CU_POINTER_ATTRIBUTE_SYNC_MEMOPS, ptr);
+                if (ret != CUDA_SUCCESS) {
+                    cuMemFree(ptr);
+                    return ret;
+                }
+            }
+
+            if (aligned_mapping)
+                out_ptr = ROUND_UP(ptr, GPU_PAGE_SIZE);
+            else
+                out_ptr = ptr;
+
+            handle->ptr = out_ptr;
+            handle->unaligned_ptr = ptr;
+            handle->size = size;
+            handle->allocated_size = allocated_size;
+
+            return CUDA_SUCCESS;
+        }
+
+        CUresult gpu_mem_free(gpu_mem_handle_t *handle)
+        {
+            CUresult ret = CUDA_SUCCESS;
+            CUdeviceptr ptr;
+
+            ret = cuMemFree(handle->unaligned_ptr);
+            if (ret == CUDA_SUCCESS)
+                memset(handle, 0, sizeof(gpu_mem_handle_t));
+
+            return ret;
+        }
+
+#if CUDA_VERSION >= 10020
+        CUresult gpu_vmm_alloc(gpu_mem_handle_t *handle, const size_t size, bool aligned_mapping, bool set_sync_memops)
+        {
+            CUresult ret = CUDA_SUCCESS;
+
+            size_t granularity, gran;
```

Comment:
we need a runtime driver version check, to deal with forward compatibility

Reply:
Done.